src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntity_SolarTower.java

```diff
@@ -599,15 +610,26 @@ public boolean checkRecipe(final ItemStack aStack) {
 			ArrayList<FluidStack> aFluids = this.getStoredFluids();
 			for (FluidStack aFluid : aFluids) {
 				if (aFluid.getFluid().equals(mColdSalt)) {
-					if (aFluid.amount >= (aTier * 1000)) {
-						this.depleteInput(FluidUtils.getFluidStack(mColdSalt, (aTier * 1000)));
-						this.addOutput(FluidUtils.getFluidStack(mHotSalt, (aTier * 1000)));
-						break;
+					int aFluidAmount = aFluid.amount;
+					if (aFluidAmount > this.mHeatLevel) {
+						this.mHeatLevel = 0;
+						log("Removed all Heat");
+						this.depleteInput(FluidUtils.getFluidStack(mColdSalt, this.mHeatLevel));
```

Comment:
use `Math.min(aFluidAmount, this.mHeatLevel)` to collapse if statement

Reply:
Done.